src/txn/txn.c

```diff
@@ -1681,8 +1500,17 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
               "durable_timestamp should not be specified for non-prepared transaction");
     }
 
-    WT_ASSERT(session,
-      !F_ISSET(txn, WT_TXN_HAS_TS_COMMIT) || txn->commit_timestamp <= txn->durable_timestamp);
+    /*
+     * Release our snapshot in case it is keeping data pinned (this is particularly important for
```

Comment:
Ah my question!

So it seems that releasing the snapshot can cause WT_UPDATE pointers to go invalid? This comment refers to cursors, so I can see how this keeps cursors valid. But I wouldn't expect all of the updates in the `mods` list to have a corresponding cursor.

That said, the old code was releasing the snapshot before walking the `mod` list. Definitely curious to learn more about this.

Reply:
Yes; a page with unresolved updates cannot be evicted, that's a separate thing. This isn't exactly my area of expertise, but in general, the snapshot is about making sure future reads the transaction might make are guaranteed to not be overtaken by eviction (or the history store), not preventing existing writes from being evicted. And, the sooner we can release the snapshot, the sooner other threads can quit worrying about future reads from this transaction.

Reply:
Yeah releasing the snapshot to free resources of other documents makes sense. But I think you answered my question by stating how a page keeps track if there are unresolved commits on it.